FinSearchUnified/BusinessLogic/ExportService.php

```diff
@@ -0,0 +1,208 @@
+     * @var FindologicArticleFactory
+     */
+    protected $findologicArticleFactory;
+
+    /**
+     * @var string
+     */
+    public $shopKey;
+
+    /**
+     * @var Category
+     */
+    protected $baseCategory;
+
+    /**
+     * @var array
+     */
+    protected $allUserGroups;
+
+    /**
+     * @var array[]
+     */
+    protected $errors = [];
+
+    public function __construct($shopkey, Category $baseCategory)
+    {
+        $this->customerRepository = Shopware()->Container()->get('models')->getRepository(Customer::class);
+        $this->articleRepository = Shopware()->Container()->get('models')->getRepository(Article::class);
+        $this->findologicArticleFactory = Shopware()->Container()->get('fin_search_unified.article_model_factory');
+        $this->shopKey = $shopkey;
+        $this->baseCategory = $baseCategory;
+        $this->allUserGroups = $this->customerRepository->getCustomerGroupsQuery()->getResult();
+    }
+
+    public function fetchTotalProductCount()
+    {
+        $countQuery = $this->articleRepository->createQueryBuilder('articles')
+            ->select('count(articles.id)')
+            ->where('articles.active = :active')
+            ->orderBy('articles.id')
+            ->setParameter('active', true);
+
+        return $countQuery->getQuery()->getScalarResult()[0][1];
+    }
+
+    public function fetchAllProducts($start, $count)
+    {
+        $articlesQuery = $this->articleRepository->createQueryBuilder('articles')
+            ->select('articles')
+            ->where('articles.active = :active')
+            ->orderBy('articles.id')
+            ->setParameter('active', true);
+
+        if ($count > 0) {
+            $articlesQuery->setMaxResults($count)->setFirstResult($start);
+        }
+
+        return $articlesQuery->getQuery()->execute();
+    }
+
+    public function fetchProductById($productId)
+    {
+         $articlesQuery = $this->articleRepository->createQueryBuilder('articles')
+            ->select('articles')
+            ->where('articles.id = :productId')
+            ->orWhere('articles.supplier = :productId')
+            ->setParameter('productId', $productId);
+
+        $shopwareArticles = $articlesQuery->getQuery()->execute();
+
+        if (count($shopwareArticles) === 0) {
+            $this->errors['general'][] = 'No article found with given ID';
+            return null;
+        }
+
+        return $shopwareArticles;
+    }
+
+    public function generateFindologicProducts($shopwareArticles, $logErrors = false)
+    {
+        $findologicArticles = [];
+
+        try {
+            /** @var Article $article */
+            foreach ($shopwareArticles as $article) {
+                $findologicArticle = self::getFindologicArticle($article, $logErrors);
+
+                if ($findologicArticle) {
+                    $findologicArticles[] = $findologicArticle;
+                }
+            }
+        } catch (Exception $e) {
+            die($e->getMessage());
```

Comment:
I really dislike that. I understand that it is easy for debugging, but in any case I do not want to throw internal stuff out there. That's just a security concern.